components/vocabularies/omim/api/src/main/java/org/phenotips/vocabulary/internal/solr/MendelianInheritanceInMan.java

```diff
@@ -243,4 +338,183 @@ private int clear()
         }
         return 1;
     }
+
+    private void index()
+    {
+        try {
+            parseOmimData();
+            loadGenes();
+            loadSymptoms(true);
+            loadSymptoms(false);
+            loadGeneReviews();
+            loadVersion();
+        } catch (NullPointerException ex) {
+            this.logger.error("Failed to prepare the OMIM index: {}", ex.getMessage(), ex);
+        }
+    }
+
+    private void parseOmimData()
+    {
+        try {
+            Reader in =
+                new InputStreamReader(new URL(OMIM_SOURCE_URL).openConnection().getInputStream(),
+                    Charset.forName(ENCODING));
+            for (CSVRecord row : CSVFormat.TDF.parse(in)) {
+                // ignore the comments, moved or removed entries
+                if (row.get(0).startsWith("#") || ("Caret").equals(row.get(0))
+                    || ("REMOVED FROM DATABASE").equals(row.get(2))) {
+                    continue;
+                }
+
+                this.crtTerm = new SolrInputDocument();
+                // set id
+                setCrtTerm(ID_FIELD, row.get(1));
+
+                // set symbol
+                setCrtTerm(SYMBOL_FIELD, SYMBOLS.get(row.get(0)));
+                // set type (multivalued)
+                for (String type : TYPES.get(row.get(0))) {
+                    setCrtTerm(TYPE_FIELD, type);
```

Comment:
`setCrtTerm` sets a value, doesn't append, so only the last set value will actually be taken into account. `crtTerm.addField` should be used instead of `crtTerm.setField`.